src/com/palmergames/bukkit/towny/Towny.java

```diff
@@ -195,12 +215,178 @@ public void onEnable() {
 		}
 	}
 
+	public void loadFoundation(boolean reload) {
+		// Before anything can continue we must load the config files, database and set the foundation for Towny.
+		loadConfig(reload);
+		// Then load the language files.
+		loadLocalization(reload);
+		// Then load the database.
+		loadDatabaseConfig(reload);
+		// Then load permissions
+		loadPermissions(reload);
+
+		// Initialize the special log4j hook logger.
+		TownyLogger.getInstance();
+
+		// Clear all objects from the TownyUniverse class.
+		townyUniverse.clearAllObjects();
+
+		// Try to load and save the database.
+		townyUniverse.loadAndSaveDatabase(TownySettings.getLoadDatabase(), TownySettings.getSaveDatabase());
+
+		// Schedule metadata to be loaded
+		MetadataLoader.getInstance().scheduleDeserialization();
+
+		// Try migrating the config and world files if the version has changed.
+		if (!TownySettings.getLastRunVersion().equals(getVersion())) {
+			ConfigMigrator migrator = new ConfigMigrator(TownySettings.getConfig(), "config-migration.json");
+			migrator.migrate();
+		}
+
+		// Loads Town and Nation Levels after migration has occured.
+		loadTownAndNationLevels();
+
+		// Run both the cleanup and backup async.
+		townyUniverse.performCleanupAndBackup();
+	}
+
+	private void loadConfig(boolean reload) {
+		// TODO: Rewrite CommentedConfiguration to take java.nio.Path instead of File.
+		// There is probably a lot of performance improvements possible for the CommentedConfiguration - Articdive.
+		try {
+			TownySettings.loadConfig(getDataFolder().toPath().resolve("settings").resolve("config.yml").toString(), getVersion());
+		} catch (IOException e) {
+			throw new TownyInitException("Failed to load the the Towny configuration.", TownyInitException.TownyError.MAIN_CONFIG, e);
+		}
+		if (reload) {
+			// If Towny is in Safe Mode (for the main config) turn off Safe Mode.
+			if (isError(TownyInitException.TownyError.MAIN_CONFIG)) {
+				removeError(TownyInitException.TownyError.MAIN_CONFIG);
+			}
+		}
+	}
+	
+	private void loadLocalization(boolean reload) {
+		Translation.loadTranslationRegistry();
+		if (reload) {
+			// If Towny is in Safe Mode (because of localization) turn off Safe Mode.
+			if (isError(TownyInitException.TownyError.LOCALIZATION)) {
+				removeError(TownyInitException.TownyError.LOCALIZATION);
+			}
+		}
+	}
+
+	private void loadDatabaseConfig(boolean reload) {
+		if (!checkForLegacyDatabaseConfig()) {
+			throw new TownyInitException("Failed to load the Towny configuration.", TownyInitException.TownyError.DATABASE_CONFIG);
```

Comment:
Bad:
- ~~Safemode is never thrown so an error on failure to load townyperms.yml results in Towny not protecting anything.~~
  - I added a tab to the first line below `nomad:`
  - Towny did not enable at all, making greifing possible.
- ~~Safemode is not thrown when the database.yml fails to load.~~
  - I changed the database save and load to squirrel.
  - Towny did not enable at all, making greifing possible.
- ~~Safemode is not thrown when the town_level is not valid yaml.~~
  - I removed a trailing ' from the namePostfix.
  - Resulted in a complete wipe of the config, Towny enabled. Versions without this logger improvement do not save a default config.yml when this happens.
- ~~Safemode is not thrown when the config is not valid yaml.~~
  - I removed a trailing ' from town.default_public ex: `default_public: 'true`
  - Resulted in a complete wipe of the config, Towny enabled. Versions without this logger improvement do not save a default config.yml when this happens.
- Breaking the yaml of the global.yml file does show a large yaml error, but otherwise has no effect (other than the global.yml overrides not being used obviously.) 
NOTE: I believe this is a part of TownyChat and not Towny itself.
- ~~Safemode is not thrown when the database cannot be loaded.~~
  - NOTE: This is analogous to database save and load having the type "squirrel", same part of the code.
  - To achieve safemode I had to change the code to return false in the loadDatabase(String) method.
  - Towny did not enable at all, making greifing possible.